### ISSUE TITLE
Vulnerability-Detector RHEL False Positives in 3.8

### ISSUE THREAD

**Issue by orlando-jamie** [wazuh/wazuh#2442]
## Bug report

**OS**
_CentOS7_

**Wazuh version**
_wazuh-manager-3.8.0-1.x86_64_
_wazuh-agent-3.8.0-1.x86_64_

**Wazuh API version**
_wazuh-api-3.8.0-1.x86_64_

**Install type**
_manager/agent_

**Install method**
_packages from https://packages.wazuh.com/3.x/yum/_

**Log sample**

```
{"timestamp":"2019-01-23T17:11:51.239+0000","rule":{"level":7,"description":"wget: Cookie injection allows malicious website to write arbitraey cookie entries into cookie jarinto cookie jar","id":"23504","firedtimes":50,"mail":true,"groups":["vulnerability-detector"],"gdpr":["IV_35.7.d"]},"agent":{"id":"001","name":"Agent_001"},"manager":{"name":"wazuh-manager"},"id":"99999999.9999999","cluster":{"name":"wazuh-cluster","node":"master_node"},"decoder":{"name":"json"},"data":{"vulnerability":{"cve":"CVE-2018-0494","title":"wget: Cookie injection allows malicious website to write arbitraey cookie entries into cookie jarinto cookie jar","severity":"moderate","published":"2018-05-06T00:00:00+00:00","state":"Fixed","cvss":{"cvss3_score":"7.100000"},"package":{"name":"wget","version":"1.14-18.el7","condition":"less than or equal 1.14-18.el7"},"advisories":"RHSA-2018:3052","cwe_reference":"CWE-20","bugzilla_reference":"https://bugzilla.redhat.com/show_bug.cgi?id=1575634","reference":"https://access.redhat.com/security/cve/CVE-2018-0494"}},"location":"vulnerability-detector"}
```

Bug description
----------------
While playing around with the new vulnerability detector, it was noticed that there are still a lot of false positive alerts while using CentOS7. 

This vulnerability is described by redhat at https://access.redhat.com/security/cve/cve-2018-0494.  According to RHSA-2018:3052, the fixed package version is wget-1.14-18.el7.x86_64.rpm. 

![screen shot 2019-01-23 at 12 21 47 pm](https://user-images.githubusercontent.com/45852956/51624748-80591800-1f09-11e9-86ec-e84ddd29ef41.png)

However, looking at the detector logic, the vulnerability-detector seems to be looking for less than or equal 1.14-18.el7. It seems for this advisory, it should really be detecting for less than 1.14-18.el7.

The offending line of code seems to be https://github.com/wazuh/wazuh/blob/63d1044b44c551781480e23b15be723c1a44a288/src/wazuh_modules/wm_vuln_detector.c#L894. The suggestion is to change the logic from "less than or equal" to just "less than".

Rationale
----------------
Since the vulnerability-detector is moving from OVAL to CVE, it should be noted that the field, AffectedRelease, is provided by RedHat to highlight the fixed version package, not the latest affected release. 

https://access.redhat.com/documentation/en-us/red_hat_security_data_api/1.0/html/red_hat_security_data_api/cve#cve_format

![screen shot 2019-01-23 at 12 27 12 pm](https://user-images.githubusercontent.com/45852956/51625110-5a804300-1f0a-11e9-9bae-b333746f7260.png)

**Comment by crolopez** [comment]
Hi @orlando-jamie,

I confirm that the error is exactly for that reason. The fix will be included in the next release of Wazuh.

You can find the development branch in tests here https://github.com/wazuh/wazuh/tree/fix-rhvuln-compare.

Thank you for your report!

Best regards.

**Comment by snaow** [comment]
Hi @orlando-jamie,

The fix has been included in latest release v3.8.1. I hope you can give it a try.
Thanks so much for the detailed report and fix suggestion.

PS: I have included credits to you in[ 3.9 changelog](https://github.com/wazuh/wazuh/blob/3.9/CHANGELOG.md)

Regards,
Pedro.
